mutex.go

```diff
@@ -130,7 +131,7 @@ func (m *Mutex) lockContext(ctx context.Context, tries int) error {
 		}
```

Comment:
I don't think it's super accurate to have `ErrAllRetriesExhausted` even -- if there's an error acquiring the lock after going through all retries, I believe an `ErrTaken` would be returned on line 130. Not sure about the situation in which line 134 executes: we'd need to have gone through all retries, where on the last retry, there was no error acquiring the lock, but the conditional check on line 117 fails.